### ISSUE TITLE
Firebase npm module incompability with new 0.37

### ISSUE THREAD

**Issue by oximer** [facebook/react-native#10958]
### Description

Until version 0.36 React native was compatible with Firebase javascript library.
https://www.npmjs.com/package/firebase

When you register a user, Firebase basically persists this information for you.
Now on version 0.37, this information is not available any more.
Thus, the user has to login every time that he opens the application.

There is no release notes that could justify this behavior. It seems to me that this is undesirable side effect.

### Reproduction

Add firebase 3.6.0 on your package.json.
Do any simple firebase example.

```
var firebase = require('firebase/app');
require('firebase/auth');
require('firebase/database');
 
var app = firebase.initializeApp({ ... });
this.FirebaseAuth = this.FirebaseApp.auth();
this.FirebaseAuth.signInWithEmailAndPassword('email@email.com', '12345678')
// ...
```

then open and close your application
try to 

```
var firebase = require('firebase/app');
require('firebase/auth');
require('firebase/database');
 
var app = firebase.initializeApp({ ... });
this.FirebaseAuth = this.FirebaseApp.auth();
this.FirebaseAuth.currentUser
```

The current user gonna be null.

### Solution

No idea.

### Additional Information

This situation is not happening when using version 0.36
I couldn't  figure out, based on release notes what affect the firebase behavior.
It affects only iOS.

* React Native version: 0.37
* Platform: iOS.
* Operating System: MacOS.
* Packages: 
  ```
    "firebase": "^3.6.0",
    "react": "15.3.2",
    "react-native": "0.37.0",
    "react-native-datepicker": "^1.4.2",
    "react-native-fbsdk": "^0.3.0",
    "react-native-fcm": "^2.3.2",
    "react-native-fetch-blob": "^0.10.0",
    "react-native-google-signin": "^0.8.1",
    "react-native-i18n": "^0.1.1",
    "react-native-image-picker": "^0.23.0",
    "react-native-keyboard-spacer": "^0.3.0",
    "react-native-swipe-list-view": "^0.3.1"
```

**Comment by hramos** [comment]
Have you filed an issue with Firebase yet?

**Comment by oximer** [comment]
Yes. I opened it. No answer yet.

Despite of the fact that this error doesn't occur when using the version 0.36, I'm checking if any other library could be affecting the compatibility between firebase + react native.

**Comment by florianbepunkt** [comment]
have you filed a github issue? If so could you post a link as I would like to follow up on this. I experience the same issue, posted it on the Firebase Google Group and referenced this issue here: https://groups.google.com/forum/#!topic/firebase-talk/h9axmktxZfY

**Comment by oximer** [comment]
@florianbepunkt Which libraries are you using on your project? The issue is also happening only on iOS?

I actually mail two firebase developers advocate about the issue. I couldn't find an appropriated topic to talk about this issue with the firebase guys.

**Comment by florianbepunkt** [comment]
@oximer I just used a blank project. only installed firebase. I only checked on ios, reveted back to 0.36.0 for now.

**Comment by oximer** [comment]
@florianbepunkt cool! 

So it's eliminates the risk that this issue was occurring due to incompatibility between firebase libraries and other libraries on the project.

**Comment by channing-google** [comment]
Hello, I'm from the Firebase team. Firebase checks if navigator.product=="ReactNative" to execute some RN-specific code. It seems that 0.37.0 changed navigator.product to be "Gecko" (at least on my machine), so Firebase thinks it's in a web browser environment.

**Comment by liorshalev01** [comment]
I experience the same problem too, while in Remote JS Debugging.

**Comment by avishayil** [comment]
Any ETA when this is going to be fixed? persisting user across app restarts stopped working as well.

**Comment by hramos** [comment]
Looks like this is being addressed by PR [#11057](https://github.com/facebook/react-native/pull/11057). See discussion at [#10881](https://github.com/facebook/react-native/issues/10881) for more details.

**Comment by starovoitovs** [comment]
Downgrading to 0.36.0 fixed the issue for now.
